src/main/java/de/hysky/skyblocker/mixin/ClientPlayNetworkHandlerMixin.java

```diff
@@ -64,4 +66,12 @@ public abstract class ClientPlayNetworkHandlerMixin {
     private void skyblocker$onParticle(ParticleS2CPacket packet, CallbackInfo ci) {
         MythologicalRitual.onParticle(packet);
     }
+
+    @Inject(method = "onBlockUpdate", at = @At("RETURN"))
+    private void skyblocker$onBlockUpdate(BlockUpdateS2CPacket packet, CallbackInfo ci) {
+        BeaconHighlighter.beaconPositions.remove(packet.getPos());
+        if(packet.getState().toString().contains("minecraft:beacon")) {
```

Comment:
Nit: code formatting, if on intellij please use `Code -> Reformat Code`

Reply:
I would also say to wrap this in a is in the end check because otherwise this is being run even in vanilla/outside of the end.

Reply:
Yeah sure, also honestly, I totally forgot IntelliJ had code formatting!
I'll get to the changes suggested after school, so in a good few hours